### PR TITLE
Use GitHub Actions for basic code checks

### PR DESCRIPTION
**Pull request by D0han**
Lets get some code checking during pull requests evaluation.  
Actual tests (ie unittests) are not executed, as those need special infra.

There is another pull request ( https://github.com/ProteGO-app/backend/pull/59 ) with reformatted code to make both PRs easily understandable and code checks green.

Proposed changes heavily base on default config supplied by GitHub.

### CODE REVIEW

**Comment by turbaszek** [comment]
Would you mind adding [pre-commit hook](https://pre-commit.com) with black (and probably others)? This will automate a lot of boring stuff including black formatting before committing changes. 

https://github.com/pre-commit/pre-commit-hooks

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - name: Lint with flake8
+      run: |
+        pip install flake8
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Check formatting with black
+      run: |
+        pip install black
```
Comment (turbaszek) [inline .github/workflows/pythonapp.yml:36]: Should we add flake8 and black to requirements? I suppose developers should run those checks locally and not only on CI.

Comment (D0han) [reply .github/workflows/pythonapp.yml:36]: There is no package to install that i can see, so there is also no place to specify such requirements. Creating something like `requirements-dev.txt` in repo root seems pretty janky.

On the other hand, i agree. We should find a way to tell developers what is needed.

**Comment by D0han** [comment]
pre-commit with black and flake8 added

**Comment by jakublipinski** [comment]
@D0han  Dzięki wielkie za wszystkie komentarze. Chcielibyśmy mieć możliwość akceptacji twoich PR. W tym celu wypełnij proszę o [oświadczenie o przekazaniu licencji](https://github.com/ProteGO-app/specs/blob/master/CONTRIBUTING.md#O%C5%9Bwiadczenie-o-udzieleniu-licencji). Po potwierdzeniu od Ministerstwa Cyfryzacji otrzymania dokumentu, zaakceptujemy PR. Każdy następny pójdzie już dużo szybciej. Zamknij ten PR jeśli nie chcesz przesłać oświadczenia. Dzięki

**Comment by turbaszek** [comment]
> pre-commit with black and flake8 added

Thanks @D0han ! Would you mind adding a note about the installation step in README.md in `Instalacja narzędzi` section?

**Comment by D0han** [comment]
@turbaszek Sure, added.

@jakublipinski Working on it.

**Comment by D0han** [comment]
I swapped flake8 and black in pre-commit config in case order matters.

@jakublipinski Done

File: .github/workflows/pythonapp.yml (hunk 1 above)
Comment (D0han) [reply .github/workflows/pythonapp.yml:33]: Any proposals?

**Pull request merged**